projects/valtimo/task/src/lib/components/task-list/task-list.component.ts

```diff
@@ -0,0 +1,408 @@
+  );
+
+  constructor(
+    private readonly configService: ConfigService,
+    private readonly documentService: DocumentService,
+    private readonly permissionService: PermissionService,
+    private readonly router: Router,
+    private readonly taskService: TaskService,
+    private readonly taskListService: TaskListService,
+    private readonly translateService: TranslateService
+  ) {
+    this.setVisibleTabs();
+  }
+
+  public paginationClicked(page: number, type: TaskListTab | string): void {
+    this.updateTaskPagination(type as TaskListTab, {page: page - 1});
+  }
+
+  public paginationSet(newSize: number): void {
+    combineLatest([this._paginationForCurrentTaskType$, this._selectedTaskType$])
+      .pipe(take(1))
+      .subscribe(([pagination, selectedTaskType]) => {
+        this.updateTaskPagination(selectedTaskType, {
+          size: Number(newSize),
+          page: this.getLastAvailablePage(
+            pagination.page,
+            Number(newSize),
+            pagination.collectionSize
+          ),
+        });
+      });
+  }
+
+  public tabChange(tab: TaskListTab | string): void {
+    this._selectedTaskType$.pipe(take(1)).subscribe(selectedTaskType => {
+      if (selectedTaskType !== tab) {
+        this.enableLoadingAnimation();
+        this._selectedTaskType$.next(tab as TaskListTab);
+        this.updateTaskPagination(tab as TaskListTab, {page: 0});
+      }
+    });
+  }
+
+  public showTask(task: Task): void {
+    this.router.navigate(['tasks', task.id]);
+  }
+
+  public openRelatedCase(event: MouseEvent, index: number): void {
+    event.stopPropagation();
+
+    this.cachedTasks$.pipe(take(1)).subscribe(cachedTasks => {
+      const currentTask = cachedTasks && cachedTasks[index];
+
+      if (currentTask && !currentTask.caseLocked) {
+        this.documentService
+          .getDocument(currentTask.businessKey)
+          .pipe(take(1))
+          .subscribe(document => {
+            this.router.navigate([
+              `/dossiers/${document.definitionId?.name}/document/${currentTask.businessKey}`,
+            ]);
+          });
+      }
+    });
+  }
+
+  public defaultTaskListFields(): void {
+    this.fields$.next(this._DEFAULT_TASK_LIST_FIELDS);
+  }
+
+  public customTaskListFields(): void {
+    const customTaskListFields = this.taskService.getConfigCustomTaskList().fields;
+
+    if (customTaskListFields) {
+      this.fields$.next(
+        customTaskListFields.map((column, index) => ({
+          key: column.propertyName,
+          label: `task-list.fieldLabels.${column.translationKey}`,
+          sortable: column.sortable,
+          ...(column.viewType && {viewType: column.viewType}),
+          ...(column.enum && {enum: column.enum}),
+        }))
+      );
+    }
+  }
+
+  public rowOpenTaskClick(task): void | boolean {
+    if (!task.endTime && !task.locked) {
+      this._taskDetail.openTaskDetails(task);
+    } else {
+      return false;
+    }
```

Comment:
```suggestion
   return !task.endTime && !task.locked ? this._taskDetail.openTaskDetails(task) : false
```

Reply:
Not absolutely necessary I just think it's a bit less cluttered this way. Dealer's choice